src/main/java/com/mentoring/ecommerce/application/service/ProductSaveService.java

```diff
@@ -0,0 +1,19 @@
+package com.mentoring.ecommerce.application.service;
+
+
+import com.mentoring.ecommerce.application.port.in.SaveProductUseCase;
+import com.mentoring.ecommerce.application.port.out.SaveProductPort;
+import com.mentoring.ecommerce.domain.Product;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
```

Comment:
I'm not sure about this annotation. Usually, I see `@Service` instead, but I keep it that way because of the _dummy_, into the `@UseCase` there is a `@Component`.

Reply:
That is actually a good point. `@Component` is more generic, and `@Service` is basically the same thing but it indicates that the class is on a service layer. Feel free to use whichever you want.